src/test/java/ro/unibuc/hello/controller/UserControllerTest.java

```diff
@@ -0,0 +1,51 @@
+package ro.unibuc.hello.controller;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import ro.unibuc.hello.data.User;
+import ro.unibuc.hello.data.UserRepository;
+import ro.unibuc.hello.dto.UserProfile;
+
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+class UserControllerTest {
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
```

Comment:
You don't need this, you should just add the SpringBootTest annotation on the class